### DISCUSSION TITLE
playlists /stations without autodj? / high cpu load from liquidsoap

### DISCUSSION THREAD

**Discussion by Tycho-X** [AzuraCast/AzuraCast#6549] [General]
Hello,

i just migrated from a simple icecast2 / ezstream setup which doesnt really need a lot of cpu/ram to AzureCast. I really like but the liquid-soap processes (3 atm) eat up between 40 and 50% of the VMs cpu while idling. Means no users connected.

I now switched off autodj but i cant find out how to attach a new playlist to station/mount without autodj/liquid soap running?

As a side note: Shouldnt the default installation be a stable release and not the rolling release?!

thx for any help....

**Comment by BusterNeece** [comment]
We use Liquidsoap as our AutoDJ, so it's responsible for gathering up playlists, handling the transition between live and AutoDJ content, and transcoding both to local mount points and to remote relays. We currently have no alternatives to Liquidsoap built-in as there frankly *are* no alternatives capable of doing what we need in that space (at least not free and open-source ones).

Your high CPU consumption may be the result of something like StereoTool or MasterMe audio processing if you have that enabled. CPU consumption of Liquidsoap is not at all affected by the total number of listeners you have; in fact, since Icecast and Shoutcast scale up rather efficiently, even large listener numbers don't tend to affect the server performance by much at all.

**Reply by Tycho-X** [reply]
Hello,

no, i didnt install any additional tools. Before i had been using the standard ezstream which basicly does a similar job without any noticable cpu load. Albeit, less perfect than autoDJ. I should be able to manually attach ezstream or similar to a mount point, right? Even i loose part of the nice GUI managment.

If this high CPU load due to liquidsoap is not typical. How to best debug this? As a sidenote, we uploaded about 2K songs into a subfolder for  one station. While uploading this songs using sftp the whole docker system did go "bongkers"/unresponsive. I had to "hard reset" the VM because the VM wouldnt even reboot. (dockerd and docker-proxy all had been "defunct".

Currently only 2 liquidsoap instances are running and the VM is on 50 to 70% CPU Load. Azurecast is the only docker container running btw. OS is Debian 11.

One more question: How to be sure to 100% to suppress transcoding? How to check if liquidsoap is doing transcoding? 

Thx,

Arnd

![screeny](https://github.com/AzuraCast/AzuraCast/assets/9759464/99be9c99-8d9a-4266-b248-a5bdfd397a6d)

![photo_2023-08-24_11-43-31](https://github.com/AzuraCast/AzuraCast/assets/9759464/dfde93b5-d47c-4d03-a09d-d6ef696e93d2)
